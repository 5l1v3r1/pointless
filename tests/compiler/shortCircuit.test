
x = false and true
y = 1 == 4 or 4 < 5