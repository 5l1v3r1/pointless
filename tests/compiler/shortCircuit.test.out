   0 [ SaveImports  ] 0       
   1 [ MakeEnv      ] 2       
   2 [ MakeNumber   ] 6       
   3 [ MakeDef      ] 0       (x)
   4 [ Jump         ] 8       (-> 12)
   5 [ LoadBool     ] 0       
   6 [ And          ]
   7 [ Jump         ] 3       (-> 10)
   8 [ LoadBool     ] 1       
   9 [ CheckBool    ] 1       
  10 [ SaveVal      ] 0       (x)
  11 [ Return       ]
  12 [ MakeNumber   ] 8       
  13 [ MakeDef      ] 1       (y)
  14 [ Jump         ] 12      (-> 26)
  15 [ MakeNumber   ] 1       
  16 [ MakeNumber   ] 4       
  17 [ Eq           ]
  18 [ Or           ]
  19 [ Jump         ] 5       (-> 24)
  20 [ MakeNumber   ] 4       
  21 [ MakeNumber   ] 5       
  22 [ LessThan     ]
  23 [ CheckBool    ] 0       
  24 [ SaveVal      ] 1       (y)
  25 [ Return       ]
  26 [ MakeObj      ]
  27 [ FieldRef     ] 10      (output)
  28 [ Jump         ] 6       (-> 34)
  29 [ Dup          ]
  30 [ LangFieldRef ] 17      (!getHead)
  31 [ Pop          ] 1       
  32 [ Jump         ] 2       (-> 34)
  33 [ LangFieldRef ] 26      (!getTail)
  34 [ Dispatch     ]
